runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -506,6 +506,12 @@ protected void rollbackStreamUnsafe(ISMRStream stream, long rollbackVersion) {
      * applied until the current tail of the stream. If Address.OPTIMISTIC
      * is given, updates will be applied to the end of the stream, and
      * upcall results will be stored in the resulting entries.
+     *
+     * When the stream is trimmed, this exception is passed up to the caller,
+     * unless the timestamp was Address.MAX, in which the entire object is
+     * reset and re-try the sync, which should pick up any checkpoint that
+     * was inserted.
+     *
```

Comment:
this sounds like a desirable behavior, indeed, but I don't see this implemented -- ?